modules/handlers/command_handlers.py

```diff
@@ -290,3 +291,137 @@ def forwarded_post_msg(update: Update, context: CallbackContext):
+    channel_id = config_map['meme']['channel_group_id'] # should be users group
+
+    abusive_message_id = MemeData.get_last_post_report(user_id=info['sender_id'], channel_id=channel_id)
+
+    info['bot'].forward_message(chat_id=chat_id,
+                                from_chat_id=channel_id,
+                                message_id=abusive_message_id)
+
+    info['bot'].sendMessage(chat_id=chat_id, text="🚨🚨 SEGNALAZIONE 🚨🚨\n\n" + info['text'])
+    info['bot'].send_message(chat_id=info['chat_id'],
+                            text="Gli admins verificheranno quanto accaduto. Grazie per la collaborazione!")
+
+    return STATE['end']
+
+# endregion
+
+# region report user
+
+def report_cmd(update: Update, context: CallbackContext) -> int:
+    """Handles the reply to the key "Report".
+    Checks the message the user wants to report
+
+    Args:
+        update (Update): update event
+        context (CallbackContext): context passed by the handler
+
+    Returns:
+        int: next state of the conversation
+    """
+    info = get_message_info(update, context)
+    if update.message.chat.type != "private":  # you can only post with a private message
+        info['bot'].send_message(
+            chat_id=info['chat_id'],
+            text="Non puoi usare quest comando ora\nChatta con @Spotted_DMI_bot in privato",
+        )
+        return STATE['end']
+
+    user_report = MemeData.get_last_user_report(user_id=info['sender_id'])
+    if user_report and user_report['is_sent']:
+        delta_time = datetime.now() - datetime.strptime(user_report['message_date'], "%Y-%m-%d %H:%M:%S.%f")
+        delta_minutes = delta_time.total_seconds()/60
+        remain_minutes = int(config_map['report_wait_mins'] - delta_minutes)
+        info['bot'].send_message(chat_id=info['chat_id'], text=f"Aspetta {remain_minutes} minuti.")
+        return STATE['end']
+
+    info['bot'].send_message(chat_id=info['chat_id'], text="Invia l'username di chi vuoi segnalare. Es. @massimobene")
+
+    return STATE['reporting_user']
+
+def report_user_msg(update: Update, context: CallbackContext) -> int:
+    """Handles the reply to the /report command after sent the @username.
+    Checks the the user wants to report, and goes to ask the reason
+
+    Args:
+        update (Update): update event
+        context (CallbackContext): context passed by the handler
+
+    Returns:
+        int: next state of the conversation
+    """
+    info = get_message_info(update, context)
+    if not check_message_type(update.message) or not info['text'].startswith('@') or info['text'].find(' ') != -1:  # the type is NOT supported
+        info['bot'].send_message(
+            chat_id=info['chat_id'],
+            text="Questo tipo di messaggio non è supportato\nÈ consentito solo username telegram. Puoi annullare il processo con /cancel")
+        return STATE['reporting_user']
+    
+    MemeData.set_user_report(user_id=info['sender_id'], evil_username=info['text'], is_sent=False)
```

Comment:
While this works, I would suggest to use the 
```python
context.user_data['...']
```
dictionary made available by the telegram library. This way you have a record in the database only if the report was actually sent. This would make useless the 'is_sent' column

Reply:
Good to know, this can simplify everything and the whole table can be removed because it will make it useless.

Reply:
Honestly idk if it makes sense to store the report in db if they are stored in admin chat.

Reply:
There are still some advantages to using a table. 
First of all, it is persistent, while the dictionary is not (to be honest, it could be made persistent, but we already have a database, so it suffices). So, wile this is fine for keeping track of some data in the conversation (since it too is not persistent), it's not ideal for actually storing data.
Having a table record could be useful to contact back the user, but arguably it is optional.

Reply:
Aight, so I will keep the table for future purposes if this is necessary.